inference-engine/thirdparty/clDNN/src/program_helpers.cpp

```diff
@@ -119,6 +119,11 @@ std::pair<bool, bool> program_helpers::are_layouts_identical(layout const& l1, l
     const auto& l2_pad = l2.data_padding;
     if (l1 == l2)
         return {true, true};
+    // Remove redundant reorder for meta-only reorder
+    if (l1.data_type == l2.data_type && l1.size == l2.size && l1.data_padding == l2.data_padding &&
+        ((l1.format.spatial_num() == 2 && l1.size.spatial[0] == 1 && l1.size.spatial[1] == 1) ||
+        ((l1.format.spatial_num() == 3 && l1.size.spatial[0] == 1 && l1.size.spatial[1] == 1 && l1.size.spatial[2] == 1))))
+        return {false, true};
```

Comment:
I think in some cases this may lead to incorrect results
What about such cases: `bsv16_fsv16 -> b_fs_yx_fsv32` or `fs_b_yx_fsv32 -> b_fs_yx_fsv16` with shape 16x32x1x1?

Reply:
possibly yes. I would add more condition about format. Let me think a little more. thanks.